rbac/management/management/commands/migrate_seeded_org_level_parents.py

```diff
@@ -0,0 +1,145 @@
+    ReplicationEvent,
+    ReplicationEventType,
+)
+from migration_tool.utils import create_relationship
+
+logger = logging.getLogger(__name__)  # pylint: disable=invalid-name
+
+
+def _child_relationship(parent_uuid: str, child_uuid: str) -> Relationship:
+    return create_relationship(
+        resource_name=("rbac", "role"),
+        resource_id=parent_uuid,
+        subject_name=("rbac", "role"),
+        subject_id=child_uuid,
+        relation="child",
+    )
+
+
+class Command(BaseCommand):
+    """Command to update the V2 parent relations for seeded roles to match their expected scope."""
+
+    help = "Update the V2 parent relations for seeded roles to match their expected scope"
+
+    def handle(self, *args, **options):
+        """
+        Update the V2 relations for system roles to reflect the appropriate scope based on their permissions.
+
+        This will not correctly handle any roles that have changed since the last time system roles were replicated to
+        relations (with seed_roles and REPLICATION_TO_RELATION_ENABLED=True).
+
+        replicator defaults to an OutboxReplicator.
+        resource_service defaults to ImplicitResourceService.from_settings().
+        """
+        replicator = OutboxReplicator()
+        resource_service = ImplicitResourceService.from_settings()
+
+        # Removing relations happens before adding them, so having the same relation in both lists is fine,
+        # and the relation will end up being present. This means we can remove the existing relations without
+        # worrying about whether we'll immediately add them back.
+        relations_to_remove: list[Relationship] = []
+        relations_to_add: list[Relationship] = []
+
+        policy_cache = GlobalPolicyIdService()
+
+        platform_default_uuid = str(policy_cache.platform_default_policy_uuid())
+        admin_default_uuid = str(policy_cache.admin_default_policy_uuid())
+
+        platform_role_for_scope = {
+            Scope.DEFAULT: platform_default_uuid,
+            Scope.ROOT: settings.SYSTEM_DEFAULT_ROOT_WORKSPACE_ROLE_UUID,
+            Scope.TENANT: settings.SYSTEM_DEFAULT_TENANT_ROLE_UUID,
+        }
+
+        admin_role_for_scope = {
+            Scope.DEFAULT: admin_default_uuid,
+            Scope.ROOT: settings.SYSTEM_ADMIN_ROOT_WORKSPACE_ROLE_UUID,
+            Scope.TENANT: settings.SYSTEM_ADMIN_TENANT_ROLE_UUID,
+        }
+
+        for role in Role.objects.public_tenant_only():
+            role_uuid = str(role.uuid)
+            target_scope = resource_service.highest_scope_for_permissions(
+                [a.permission.permission for a in role.access.all()]
+            )
+
+            role_desc = f"{repr(role.name)} (id={role.id})"
+
+            if role.platform_default:
+                logger.info(
+                    f"Rebinding role {role_desc} to have platform-default parent role for scope {target_scope.name}."
+                )
+
+                relations_to_remove.append(
```

Comment:
If we remove not only the default one but all scopes, that would support the case if a team update its scope to root and then update it again to tenant.
Could you help confirming removing non exist relationship is OK?

Reply:
Yes, I considered that; I figured that was stepping on the toes of [RHCLOUD-41067](https://issues.redhat.com/browse/RHCLOUD-41067) and/or [RHCLOUD-40813](https://issues.redhat.com/browse/RHCLOUD-40813).